### PR TITLE
Support for logging to console on tendermint com server

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

Adds support for logging to console on the tendermint com server

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes
- [ ] Public-facing documentation has been updated with the changes affected by this PR. Even if the provided contents are not in their final form, all significant information must be included.
- [ ] Any backwards-incompatible/breaking change has been clearly documented in the upgrading document.

### CODE REVIEW

**Review by angrybayblade — COMMENTED** [review]
Tested this locally with trader

File: deployments/Dockerfiles/tendermint/tendermint.py [hunk 1]
```diff
@@ -117,47 +118,48 @@ def build_node_command(self, debug: bool = False) -> List[str]:
         ]
         if debug:
             cmd.append("--log_level=debug")
-
         if self.home is not None:  # pragma: nocover
             cmd += ["--home", self.home]
         return cmd
 
     @staticmethod
-    def get_node_command_kwargs(monitoring: bool = False) -> Dict:
```
Comment (DavidMinarsch) [inline deployments/Dockerfiles/tendermint/tendermint.py:126]: Hm this approach was more generic. Why are we forcing the approach now?

File: deployments/Dockerfiles/tendermint/tendermint.py [hunk 2]
```diff
@@ -174,35 +176,64 @@ def init(self) -> None:
         cmd = self._build_init_command()
         subprocess.call(cmd)  # nosec
 
-    def start(self, start_monitoring: bool = False, debug: bool = False) -> None:
-        """Start a Tendermint node process."""
-        self._start_tm_process(start_monitoring, debug)
-        if start_monitoring:
-            self._start_monitoring_thread()
+    def _monitor_tendermint_process(
```
Comment (DavidMinarsch) [inline deployments/Dockerfiles/tendermint/tendermint.py:179]: Is this a complete copy from below?

**Review by DavidMinarsch — COMMENTED** [review]
Not sure what the reason was for removing the option to select the type of monitoring. Isn't it simpler to just modify the defaults?

**Comment by solarw** [comment]
@angrybayblade 
tendermint already works well with printing lofs to stdout
i just set log file env variable to /dev/null to avoid on-disk files usage.

so now i can just set WRITE_TO_LOG=false?  and it will stop writing logs to file?

**Comment by angrybayblade** [comment]
> @angrybayblade tendermint already works well with printing lofs to stdout i just set log file env variable to /dev/null to avoid on-disk files usage.
> 
> so now i can just set WRITE_TO_LOG=false? and it will stop writing logs to file?

Yes

File: deployments/Dockerfiles/tendermint/tendermint.py (hunk 2 above)
Comment (angrybayblade) [reply deployments/Dockerfiles/tendermint/tendermint.py:179]: Yes

**Review by solarw — APPROVED** [review]
LGTM

File: deployments/Dockerfiles/tendermint/tendermint.py (hunk 1 above)
Comment (DavidMinarsch) [reply deployments/Dockerfiles/tendermint/tendermint.py:126]: No monitoring was a flag that got passed down. So it was possible to configure.

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]
What exactly is this PR trying to achieve?

It does change some default behaviour and adds support to configure via env variable and moves some code around. Can we be specific on what we actually want from this PR?

**Review by DavidMinarsch — APPROVED** [review]

**Review by solarw — APPROVED** [review]
lgtm

**Review by 0xArdi — APPROVED** [review]

**Review by 0xArdi — APPROVED** [review]

**Pull request merged**